execution-frameworks/Invoke-AtomicRedTeam/Invoke-AtomicRedTeam.ps1

```diff
@@ -0,0 +1,147 @@
+.EXAMPLE
+Generate the Atomic Tests For A Given Technique, don't execute.
+Invoke-AtomicTest $T1117 -GenerateOnly
+.EXAMPLE
+Execute the Atomic Tests For A Given Technique
+$T1117 = Get-AtomicTechnique -Path ..\..\atomics\T1117\T1117.yaml
+Invoke-AtomicTest $T1117
+.NOTES
+This script converts Atomic Tests Expressed in YAML into PowerShell Objects.
+
+.LINK
+Blog: http://subt0x11.blogspot.com/2018/08/invoke-atomictest-automating-mitre-att.html
+Github repo: https://github.com/redcanaryco/atomic-red-team
+
+#>
+
+function Confirm-Dependencies {
+
+    if (Get-Module -ListAvailable -Name 'powershell-yaml') {
+        Write-Host "PowerShell-Yaml Module Exists. Good To Go." -Foreground Green
+    }
+    else {
+        $Title = "PowerShell-Yaml Is Not Installed"
+        $Message = "Do you want to Install?"
+        $Yes = New-Object System.Management.Automation.Host.ChoiceDescription "&Y-Yes", "Yes"
+        $No = New-Object System.Management.Automation.Host.ChoiceDescription "&N-No", "No"
+        $Options = [System.Management.Automation.Host.ChoiceDescription[]]($Yes, $No)
+        $Result = $Host.ui.PromptForChoice($Title, $Message, $Options, 0)
+
+        switch ($Result) {
+            0 { Install-Module -Name powershell-yaml }
+            1 { exit }
+        }
+    }
+}
+
+function Get-AtomicTechnique {
+    [CmdletBinding()]
+    Param(
+        [string]
+        $Path
+    )
+    # Returns A HashTable For Each File Passed In
+    BEGIN { }
+    PROCESS {
+        foreach ($File in $Path) {
+            $parsedYaml = (ConvertFrom-Yaml (Get-Content $File -Raw ))
+            Write-Output $parsedYaml
+        }
+    }
+    END { }
+}
+
+function Invoke-AtomicTest {
+    [CmdletBinding()]
+    Param(
+        [System.Collections.Hashtable]
+        $AtomicTechnique,
+
+        [switch]
+        $GenerateOnly
+    )
+    BEGIN {}
+    PROCESS {
+        foreach ($Technique in $AtomicTechnique) {
+            $AtomicTest = $Technique.atomic_tests
+
+            foreach ($Test in $AtomicTest) {
+                #Only Process Windows Tests For Now
+                if (-Not $Test.supported_platforms.Contains('windows')) {
+                    return
+                }
+
+                #Reject Manual Tests
+                if ($Test.executor.name.Contains('manual')) {
+                    return
+                }
+
+                Write-Host ("[********BEGIN TEST*******]`n" +
+                    $Technique.display_name.ToString(), $Technique.attack_technique.ToString() )
+                Write-Host $Test.name.ToString()
+                Write-Host $Test.description.ToString()
+
+                $finalCommand = $Test.executor.command
+                if ($Test.input_arguments.Count -gt 0) {
+                    #Replace InputArgs with default values
+                    $InputArgs = [Array]($Test.input_arguments.Keys).Split(" ")
+                    $InputDefaults = [Array]( $Test.input_arguments.Values | ForEach-Object {$_.default }).Split(" ")
```

Comment:
I expanded your use of `% { }` to `ForEach-Object { }`. In scripts, you should avoid using the shortcuts for readability.

Those short cuts (`?`, `%`, `iex`) are really only meant to be used in the interactive console.